### PR TITLE
Remove redundant softmax callbacks

### PR DESCRIPTION
**Pull request by CasualAutopsy**
Removes unneeded softmax calculations since nearly every sampler applies it before base sampler calculations.

For the sake of keeping old presets the same, I've added the boolean `norm_minp` which makes it so that Min P calculates token softmax before applying Min P itself.

Samplers that have had redundant softmax calculations removed:
- `Top N Sigma`
- `Smooth Sampling`
- `Dyna. Temp.`

### CODE REVIEW

**Comment by LostRuins** [comment]
i'm not really seeing the benefit here, performance seems exactly the same. I don't think I want to risk changing sampler behavior for basically no gain.

**Comment by CasualAutopsy** [comment]
> i'm not really seeing the benefit here, performance seems exactly the same. I don't think I want to risk changing sampler behavior for basically no gain.

What about Just removing the one right after Top N-Sigma then? LlamaCpp did a commit for that not too long ago, so doing so would make it more aligned with upstream.

**Comment by LostRuins** [comment]
Why though? It's important for the output logits to be softmaxed because the next sampler in the chain might expect it to be.
